storage/innobase/handler/ha_innodb.cc

```diff
@@ -3639,6 +3639,60 @@ static uint innobase_partition_flags()
 	return (0);
 }
 
+/** Return the minimum buffer pool size based on page size */
+static inline ulint min_buffer_pool_size()
+{
+  ulint s= (BUF_LRU_MIN_LEN + BUF_LRU_MIN_LEN / 4) * srv_page_size;
+  /* buf_pool_chunk_size minimum is 1M, so round up to a multiple */
+  if (s % (1 << 20))
+     s+= (1 << 20) - (s % (1 << 20));
+  return(s);
```

Comment:
This could be:
```c++
  if (ulint remainder= s % (1<< 20))
    s+= (1 << 20) - remainder;
  return s;
```
However, it would be even better to use pure arithmetics without any condition:
```c++
  ulint alignment= 1U << 20;
  return UT_CALC_ALIGN(s, alignment);
```